icu4c/source/common/uloc.cpp

```diff
@@ -1465,6 +1465,25 @@ uloc_openKeywords(const char* localeID,
 static const char i_default[] = {'i', '-', 'd', 'e', 'f', 'a', 'u', 'l', 't'};
 #define I_DEFAULT_LENGTH UPRV_LENGTHOF(i_default)
 
+static int32_t _preprocessLocaleID(const char* inLocaleID, char* outBuffer, int32_t bufferCapacity, UErrorCode* err) {
```

Comment:
In a function that takes a UErrorCode, normally the first line should be something like
```
if (U_FAILURE(*err)) { return 0; }
```

Reply:
Is it necessary to copy+modify the locale ID string? Wouldn't it be easier to parameterize the parser for allowing an underscore despite singleton extensions?

Reply:
Is it necessary to copy+modify the locale ID string? Wouldn't it be easier to parameterize the parser for allowing an underscore despite singleton extensions?

Reply:
I considered parameterizing the parser, but the logic that would need to change based on the parameter was about six steps away from here in the call chain, and I didn't want to have to change the signatures of all the intervening functions to account for this.  It seemed simpler to just fix up the string at the beginning.